src/tx/applyTx/utils.js

```diff
@@ -27,9 +26,24 @@ const checkInsAndOuts = (tx, state, unspentFilter) => {
 
   const insValues = inputTransactions.reduce(groupValuesByColor, {});
   const outsValues = tx.outputs.reduce(groupValuesByColor, {});
-  const colors = Object.keys(insValues);
+  const colors = uniq([
+    ...Object.keys(insValues),
+    ...Object.keys(outsValues),
+  ]).map(Number);
+  const gasPrice = bridgeState.gasPrice || 0;
+  const gas = Math.max(0, tx.outputs.length * 20000 - tx.inputs.length * 10000);
+
   for (const color of colors) {
-    if (!isEqual(insValues[color], outsValues[color])) {
+    if (color === 0 && gas > 0 && gasPrice > 0 && tx.type === Type.TRANSFER) {
+      const inputValue = insValues[color] || 0;
+      const outputValue = outsValues[color] || 0;
+      const txGasPrice = (inputValue - outputValue) / gas;
+      if (txGasPrice > gasPrice) {
+        throw new Error(`Tx overpriced`); // is this a really reason to decline tx?
```

Comment:
no, i think it is totally legit to have overpriced transactions, even desirable.
i was under the assumption that the transaction would have a dedicated field called gasPrice. then the field should match the calculation exactly. but seems not necessary.